mne/viz/misc.py

```diff
@@ -625,3 +626,192 @@ def plot_dipole_amplitudes(dipoles, colors=None, show=True):
+    """
+    axes = [axes] if not isinstance(axes, (list, tuple, np.ndarray)) else axes
+    for ax in axes:
+        if grid:
+            ax.grid(zorder=0)
+        for key in remove_spines:
+            ax.spines[key].set_visible(False)
+
+
+def _log_ticks(lims):
+    """Create approximately logarithmically spaced ticks between lims."""
+    lims = np.array(lims)
+    lims[0] = 0.01 if lims[0] == 0 else lims[0]
+    ticks = list()
+    for exp in range(int(np.floor(np.log10(lims[0]))),
+                     int(np.floor(np.log10(lims[1]))) + 1):
+        ticks += (np.array([1, 2, 4]) * (10 ** exp)).tolist()
+    ticks = np.round(np.array(ticks)).astype(int)
+    ticks = ticks[(ticks >= lims[0]) & (ticks <= lims[1])]
+    return ticks
+
+
+def plot_filter(h, sfreq, freq=None, gain=None, title=None, color='#1f77b4',
+                flim=None, alim=(-60, 10), show=True):
+    """Plot properties of a filter
+
+    Parameters
+    ----------
+    h : tuple or ndarray
+        Can be a (b, a) tuple, 1D ndarray (for FIR filter) or 2D ndarray
+        for SOS filter.
+    freq : array-like or None
+        The ideal response frequencies to plot.
+        If None, do not plot the ideal response.
+    gain : array-like or None
+        The ideal response gains to plot.
+        If None, do not plot the ideal response.
+    title : str | None
+        The title to use. If None, deteremine the title based on the
+        type of the system.
+    color : color object
+        The color to use.
+    flim : tuple or None
+        If not None, the x-axis frequency limits (Hz) to use.
+    alim : tuple
+        The y-axis amplitude limits (dB) to use.
+    show : bool
+        Show figure if True.
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure containing the plots.
+
+    Notes
+    -----
+    .. versionadded:: 0.13
+    """
+    from scipy.signal import freqz, group_delay
+    import matplotlib.pyplot as plt
+    if h.ndim == 2:  # second-order sections
+        from scipy.signal import sosfilt
+        H = np.ones(512, np.complex128)
+        gd = np.zeros(512)
+        for section in h:
+            f, this_H = freqz(section[:3], section[3:])
+            H *= this_H
+            gd += group_delay((section[:3], section[3:]))[1]
+        n = estimate_ringing_samples(h)
+        delta = np.zeros(n)
+        delta[0] = 1
+        h = sosfilt(h, delta)
+        title = 'SOS (IIR) filter' if title is None else title
+    else:
+        f, H = freqz(h)
+        gd = group_delay((h, [1.]))[1]
+        title = 'FIR filter' if title is None else title
+    gd /= sfreq
+    fig, axes = plt.subplots(3)  # eventually axes could be a parameter
+    t = np.arange(len(h)) / sfreq
+    axes[0].plot(t, h, color=color)
+    axes[0].set(xlim=t[[0, -1]], xlabel='Time (sec)',
+                ylabel='Amplitude h(n)', title=title)
+    f *= sfreq / (2 * np.pi)
+    axes[1].semilogx(f, 10 * np.log10((H * H.conj()).real), color=color,
```

Comment:
I believe this should be `axes[1].plot` -- otherwise you don't see anything for highpass filters because of the scale. Unless you have somehow taken care of that in some other way ...